src/request.ts

```diff
@@ -1122,18 +1211,27 @@ class DatastoreRequest {
       reqOpts.transaction = this.id;
     }
 
+    if (
+      isTransaction ||
+      (reqOpts.readOptions && reqOpts.readOptions.newTransaction)
+    ) {
+      if (reqOpts.readOptions && reqOpts.readOptions.readConsistency) {
+        callback(
+          new Error('Read consistency cannot be specified in a transaction.')
+        );
+        return;
+      }
+      if (reqOpts.readOptions && reqOpts.readOptions.readTime) {
+        callback(new Error('Read time cannot be specified in a transaction.'));
+        return;
+      }
+    }
     if (
       isTransaction &&
       (method === 'lookup' ||
         method === 'runQuery' ||
         method === 'runAggregationQuery')
     ) {
-      if (reqOpts.readOptions && reqOpts.readOptions.readConsistency) {
-        throw new Error(
-          'Read consistency cannot be specified in a transaction.'
-        );
-      }
-
```

Comment:
This is just moved up to also be included for new transaction calls. Also, the error was not propogating up and needs to be wrapped in a callback.